### PR TITLE
NET-466: tracker drops duplicate id connections

### PR DESCRIPTION
**Pull request by juslesan**
- To avoid confusion when sessionIds are not used, users should not be able to spin up multiple nodes with the same Ethereum private key
- Tracker now rejects connections with the same nodeId (ethereum address)

### CODE REVIEW

File: packages/network/test/integration/browser-ws-endpoint.test.ts [hunk 1]
```diff
@@ -93,4 +93,23 @@ describe('ws-endpoint', () => {
...
+
+            await client1.connect(server.getUrl(), PeerInfo.newTracker('server'))
+            try {
+                await client2.connect(server.getUrl(), PeerInfo.newTracker('server'))
+            } catch (err) {
+                expect(err).toContain('private key')
+            }
+            await client1.stop()
+            await client2.stop()
+            await server.stop()
```
Comment (harbu) [inline packages/network/test/integration/browser-ws-endpoint.test.ts]: Probably broke indentation here, but to ensure cleanup happens: try-finally.
```suggestion
            try {
              await client1.connect(server.getUrl(), PeerInfo.newTracker('server'))
              try {
                  await client2.connect(server.getUrl(), PeerInfo.newTracker('server'))
              } catch (err) {
                  expect(err).toContain('private key')
              }
          } finally {
              await Promise.allSettled([
                  client1.stop(),
                  client2.stop(),
                  server.stop()
              ])
          }
```

File: packages/network/test/integration/ws-endpoint.test.ts [hunk 2]
```diff
@@ -97,4 +97,23 @@ describe('ws-endpoint', () => {
...
+
+            await client1.connect(server.getUrl(), PeerInfo.newTracker('server'))
+            try {
+                await client2.connect(server.getUrl(), PeerInfo.newTracker('server'))
+            } catch (err) {
+                expect(err).toContain('private key')
+            }
+            await client1.stop()
+            await client2.stop()
+            await server.stop()
```
Comment (harbu) [inline packages/network/test/integration/ws-endpoint.test.ts:114]: try-finally cleanup here a well

File: packages/network/test/integration/browser-ws-endpoint.test.ts [hunk 3]
```diff
@@ -93,4 +93,23 @@ describe('ws-endpoint', () => {
...
+            const client1 = new BrowserClientWsEndpoint(PeerInfo.newNode('client'))
+            const client2 = new BrowserClientWsEndpoint(PeerInfo.newNode('client'))
+
+            const server = await startServerWsEndpoint('127.0.0.1', 38482, PeerInfo.newNode('server'))
+
+            await client1.connect(server.getUrl(), PeerInfo.newTracker('server'))
+            try {
+                await client2.connect(server.getUrl(), PeerInfo.newTracker('server'))
+            } catch (err) {
+                expect(err).toContain('private key')
```
Comment (harbu) [inline packages/network/test/integration/browser-ws-endpoint.test.ts]: Maybe more fitting given the context of the test
```suggestion
                expect(err).toContain('Duplicate nodeId detected')
```

File: packages/network/test/integration/ws-endpoint.test.ts [hunk 4]
```diff
@@ -97,4 +97,23 @@ describe('ws-endpoint', () => {
...
+            const client1 = new NodeClientWsEndpoint(PeerInfo.newNode('client'))
+            const client2 = new NodeClientWsEndpoint(PeerInfo.newNode('client'))
+
+            const server = await startServerWsEndpoint('127.0.0.1', 38482, PeerInfo.newNode('server'))
+
+            await client1.connect(server.getUrl(), PeerInfo.newTracker('server'))
+            try {
+                await client2.connect(server.getUrl(), PeerInfo.newTracker('server'))
+            } catch (err) {
+                expect(err).toContain('private key')
```
Comment (harbu) [inline packages/network/test/integration/ws-endpoint.test.ts]:
```suggestion
                expect(err).toContain('Duplicate nodeId detected')
```

**Review by harbu — CHANGES_REQUESTED** [review]

File: packages/network/test/integration/browser-ws-endpoint.test.ts [hunk 5]
```diff
@@ -93,4 +93,23 @@ describe('ws-endpoint', () => {
...
+        it('Duplicate connection is closed', async () => {
+            const client1 = new BrowserClientWsEndpoint(PeerInfo.newNode('client'))
+            const client2 = new BrowserClientWsEndpoint(PeerInfo.newNode('client'))
+
+            const server = await startServerWsEndpoint('127.0.0.1', 38482, PeerInfo.newNode('server'))
+
+            await client1.connect(server.getUrl(), PeerInfo.newTracker('server'))
+            try {
+                await client2.connect(server.getUrl(), PeerInfo.newTracker('server'))
+            } catch (err) {
```
Comment (harbu) [inline packages/network/test/integration/browser-ws-endpoint.test.ts]:
```suggestion
                await client2.connect(server.getUrl(), PeerInfo.newTracker('server'))
                fail('did not throw')
            } catch (err) {
```

File: packages/network/test/integration/ws-endpoint.test.ts [hunk 6]
```diff
@@ -97,4 +97,23 @@ describe('ws-endpoint', () => {
...
+    describe('Duplicate connections from same nodeId are closed', () => {
+        it('Duplicate connection is closed', async () => {
+            const client1 = new NodeClientWsEndpoint(PeerInfo.newNode('client'))
+            const client2 = new NodeClientWsEndpoint(PeerInfo.newNode('client'))
+
+            const server = await startServerWsEndpoint('127.0.0.1', 38482, PeerInfo.newNode('server'))
+
+            await client1.connect(server.getUrl(), PeerInfo.newTracker('server'))
+            try {
+                await client2.connect(server.getUrl(), PeerInfo.newTracker('server'))
```
Comment (harbu) [inline packages/network/test/integration/ws-endpoint.test.ts]:
```suggestion
                await client2.connect(server.getUrl(), PeerInfo.newTracker('server'))
                fail('did not throw')
```

**Review by harbu — CHANGES_REQUESTED** [review]

**Comment by timoxley** [comment]
I think this will cause problems with the brubeck client model. If every client spins up its own node, with this change, you won't be able to use  multiple clients with the same auth credentials, right?

**Review by harbu — APPROVED** [review]

**Pull request merged**